QEfficient/generation/text_generation_inference.py

```diff
@@ -118,15 +140,25 @@ def check_batch_size_and_num_prompts(prompt, prompts_txt_file_path, batch_size)
         prompt = read_prompts_txt_file(prompts_txt_file_path)
     if isinstance(prompt, str):
         prompt = [prompt]
-
-    num_prompts = len(prompt)
-    if batch_size > 1:
-        assert (
-            batch_size == num_prompts
-        ), f"Mismatch between number of prompts {num_prompts} and batch size {batch_size}; please pass correct input argument"
     return prompt
 
 
+def check_batch_size_and_num_prompts(prompt: List[str], batch_size: int):
```

Comment:
rename to `fix_prompts_and_get_num_iters`.
You don't need to return prompt, as list are passed by reference.
return only `num_iterations`